packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -136,22 +169,47 @@ export const messageThreadSelector: MessageThreadSelector = createSelector(
             message.clientMessageId !== undefined
         )
         .filter((message) => message.content && message.content.message !== '') // TODO: deal with deleted message and remove
-        .map((message) =>
-          memoizedFn(
+        .map((message) => {
+          /** logic: Looking at message A, how do we know it's read number?
+           * Assumption: if user read the latest message, user has read all messages before that
+           * ReadReceipt behaviour: read receipt is only sent to the last message
+           * if participant read a message that is sent later than message A, then the participant has read message A
+           * how do we check if the message is sent later than message A?
+           * we compare if the readon timestamp is later than the message A sent time
+           * if last read message id is not equal to message A's id, check the read on time stamp.
+           * if the last read message is read after the message A is sent, then user should have read message A as well */
+
+          let readNumber = 0;
```

Comment:
This logic is working well to calculate the readNum, thanks for implementing this part! In addition, there could be some perf improvements but not necessary for this PR - so we don't need 2 loops O(n2) to get readNum:
1. Iterate messages from bottom to top and the last readMessage id for users, remember each last read Message Id for each user
2. when calculate readNum using this lastMessageID, just need to do O(20*n) because max participants number of enable readReceipt will be only 20

This improved logic doesn't have to be in this PR cause this one is working very well. Let's open another following PR after this one gets checked in

Reply:
created work item for this 
https://skype.visualstudio.com/SPOOL/_workitems/edit/2762970